packages/@uppy/companion/src/server/Uploader.js

```diff
@@ -188,13 +198,30 @@ class Uploader {
         // The stream will then typically come from a "Transfer-Encoding: chunked" response
         await this._downloadStreamAsFile(this.readStream)
       }
-      if (this.uploadStopped) return
+      if (this.uploadStopped) return undefined
 
       const { url, extraData } = await Promise.race([
         this._uploadByProtocol(),
         // If we don't handle stream errors, we get unhandled error in node.
         new Promise((resolve, reject) => this.readStream.on('error', reject)),
       ])
+      return { url, extraData }
+    } finally {
+      logger.debug('cleanup', this.shortToken)
+      if (this.readStream && !this.readStream.destroyed) this.readStream.destroy()
+      if (this.tmpPath) unlink(this.tmpPath).catch(() => {})
+    }
+  }
+
+  /**
+   *
+   * @param {import('stream').Readable} stream
+   */
+  async tryUploadStream (stream) {
+    try {
+      const ret = await this.uploadStream(stream)
+      if (!ret) return
```

Comment:
I'd like to pentition for us to start using nullish comparisons instead of falsy ones. Using falsy can be a footgun (because `0` and the empty string are falsy), and imho it's more readable and is more self-explanatory.
```suggestion
      if (ret == null) return
```

Reply:
I agree that nullish comparison is better when the function is expected to sometimes return a value where we consider 0 or '' to be valid values. In this case,`if (ret == null) return` would not return if `ret` by mistake is 0 or '', and it would continue on to the next line and crash, so I feel that `!ret` is "safer", as it will match more strictly. (it will return on more invalid values). do I make any sense?

Reply:
I disagree it's "safer", I would argue the opposite: if the function returns `0` or an empty string it's almost certainly an error that we want to fix so I think a crash is more helpful than silently swallowing the type mismatch. Also, it wouldn't help if the function returns e.g. `1`.
Anyway, if we really cared this kind of safety, we would write TypeScript (or make JSDoc mandatory) so I don't feel strongly about it, use the syntax that makes the more sense to you.

Reply:
Actually when I think about it if we use `if (ret == null) return`, and ret is 0, then it would:
```js
const { url, extraData } = 0
```
so it wouldn't crash, but it would `emitSuccess(undefined, undefined)`
Honestly I'm not sure what is best in this case, I just felt that `!ret` would be less likely to lead to strange behavior if ret was not an object, which is what we want in the success-case. Is there any way we can enforce `x != null` instead of `!ret` with eslint?

Reply:
I suppose we could turn on an option to force to write `== true` for all conditions, but that would make checking for actual boolean value needlessly verbose, so I would be hesitant to turn on such option...